### PR TITLE
Addon-docs: Support non-story exports in module format

### PR DESCRIPTION
**Pull request by shilman**
Issue: #7071 

## What I did

Added includeStories/excludeStories component metadata to make it easy to control which file exports get treated as stories. (So that story files can also export data)

## How to test

```
yarn jest --testPathPattern start.test.js
```

### CODE REVIEW

File: lib/core/src/client/preview/start.js [hunk 1]
```diff
@@ -283,7 +297,7 @@ export default function start(render, { decorateStory } = {}) {
         );
       }
 
-      const { default: meta, ...examples } = fileExports;
+      const { default: meta, ...stories } = fileExports;
```
Comment (tmeasday) [inline lib/core/src/client/preview/start.js]: exports ?

File: lib/core/src/client/preview/start.js [hunk 2]
```diff
@@ -21,6 +21,20 @@ const classes = {
...
 };
 
+function matches(storyKey, arrayOrRegex) {
+  if (Array.isArray(arrayOrRegex)) {
+    return arrayOrRegex.includes(storyKey);
+  }
+  return storyKey.match(arrayOrRegex);
+}
+
+export function includeStory(storyKey, { includeStories, excludeStories }) {
```
Comment (tmeasday) [inline lib/core/src/client/preview/start.js]: exportIsStory?

**Review by tmeasday — APPROVED** [review]
LGTM with some minor renamings to aid clarity

**Comment by bdombro** [comment]
Hint: To use the new feature you can now allowList using `includeStories: ['myStory1', ...]` or blockList using `excludeStores: ['myNonStoryExport', ...]` in `export default`

**Pull request merged**